metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -87,6 +100,21 @@ class LDAPSourceConfig(ConfigModel):
         default=20, description="Size of each page to fetch when extracting metadata."
     )
 
+    # default mapping for attrs
+    attrs_mapping: Dict[str, Any] = {}
+
+
+def guess_person_ldap(attrs: Dict[str, Any], config: LDAPSourceConfig) -> Optional[str]:
+    """Determine the user's LDAP based on the DN and attributes."""
+    if config.attrs_mapping["urn"] in attrs:
+        return attrs[config.attrs_mapping["urn"]][0].decode()
```

Comment:
So since we default `config.attrs_mapping["urn"]` to `sAMAccountName`, the `urn` key will always be present in the `attrs_mapping` dict (whether user specifies `urn` mapping in the recipe or not.) Also, this function returning `None` is current behavior but now we give users more flexibility to pick a more reliable attribute in their org to minimize the chances this function will return none.